File: C_Workspace/C_CheckList/Functions/Recursion/4.c
//C program to find sum of all digits using recursion.
#include <stdio.h>

int sum(int);

int sum(int n)
{
    int count=n%10;
    if(n==0)
    {
        return 0;
    }
    else
    {
        return count+sum(n/10);
    }
}

int main()
{
    int x;
    printf("Enter the number:");
    scanf("%d",&x);
    if(x==0)
    {
    printf("The sum of digits in %d are 0",x);
    }
    else
    {
        printf("The sum of digits in %d are %d",x,sum(x));
    }
    return 0;
}

File: Python_Workspace/Python_Tasks/16_05/4.py
#Friendly pair 
a=int(input("Enter the 1st number:"))
b=int(input("Enter the 2nd number:"))
x=y=0
for i in range (1,a+1):
  if a%i==0:
    x=x+i
print(x)
for j in range (1,b+1):
  if b%j==0:
    y=y+j
print(y)
p=x/a
q=y/b
print(p,q)
if p==q:
  print(a,"and",b,"are Friendly pair")
else:
  print(a,"and",b,"are not Friendly pair")
